### ISSUE TITLE
Hi,is there some documents for it ?

### ISSUE THREAD

**Issue by avayayu** [vergl4s/botcoin#1]
about some design idea?

**Comment by vergl4s** [comment]
No, not really. This is still in alpha and basic things change everyday. Once it's in beta I'll add some documentation. Will keep this issue open until documentation exists.

**Comment by avayayu** [comment]
ok. I have read source code recently. I have a question about the event sequence of backtestMarkData. What's the event such like 'before_open','on_open','on_close','after_close' mean?
what's the different between ''before_open'' and 'on_open'?

**Comment by vergl4s** [comment]
OK so like you said these events are used for backtesting purposes. 'before_open' and 'after_close' are called a single time and no parameter is passed (you should only use 'def before_open(self):'). I use them for any strategy that requires to pre-compile a list. The others are called when the market is open, to simulate the different stages e.g. open, low, high, close.

This setup isn't working out well, and I'm not even using it. You can just ignore them and use 'on_close' or 'after_close'. For more examples check out the test strategies at /tests/test-strategies.
